transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -140,91 +134,139 @@ public String name() {
     public ChannelHandlerContext fireChannelRegistered() {
         AbstractChannelHandlerContext next = findContextInbound(MASK_CHANNEL_REGISTERED);
         if (next.executor().inEventLoop()) {
-            if (next.invokeHandler()) {
-                try {
-                    final ChannelHandler handler = next.handler();
-                    ((ChannelInboundHandler) handler).channelRegistered(next);
-                } catch (Throwable t) {
-                    next.invokeExceptionCaught(t);
+            next.invokeFireChannelRegistered();
+        } else {
+            next.executor().execute(next::invokeFireChannelRegistered);
+        }
+        return this;
+    }
+
+    private void invokeFireChannelRegistered() {
```

Comment:
Here you are undoing the pipeline stack depth reduction from https://github.com/netty/netty/pull/14705

Can we accomplish the pending bytes change while preserving that the channel handler context calls the `channel*` methods directly from its `fire*` methods?